src/Psalm/Type/Atomic/TKeyedArray.php

```diff
@@ -663,8 +663,22 @@ public function getList(): TList
      */
     private function escapeAndQuote($name)
     {
-        if (is_string($name) && ($name === '' || preg_match('/[^a-zA-Z0-9_]/', $name))) {
-            $name = '\'' . str_replace("\n", '\n', addslashes($name)) . '\'';
+        if (is_string($name)) {
+            $quote = false;
+
+            if ($name === '' || preg_match('/[^a-zA-Z0-9_]/', $name)) {
+                $quote = true;
+            }
+
+            if (preg_match('/^[1-9][0-9]*$/', $name)
+                && (string)(int) $name !== $name // overflow occured
+            ) {
+                $quote = true;
+            }
```

Comment:
This is to distinguish strings that look like integers, but won't be converted to integers by PHP (e.g. any that is greater than `PHP_INT_MAX`);

Reply:
What about numbers smaller than PHP_INT_MIN?
```suggestion
            if (preg_match('/^-?[1-9][0-9]*$/', $name)
                && (string)(int) $name !== $name // overflow occured
            ) {
                $quote = true;
            }
```

Am I understanding this correctly that quoting done here is only for users to distinguish numeric-string keys from ints?

Reply:
> Am I understanding this correctly that quoting done here is only for users to distinguish numeric-string keys from ints?

Correct.